pkg/operator/reports.go

```diff
@@ -91,8 +91,26 @@ func (op *Reporting) syncReport(logger log.FieldLogger, key string) error {
 		}
 		return err
 	}
+
 	sr := report.DeepCopy()
 
+	// check if the report is expired
+	if report.DeletionTimestamp == nil {
+		if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
```

Comment:
I would just deflate this into a single conditional. Easier to read imo.

```suggestion
        // check if the report we're currently processing in the queue has reached the
        // configuring rention period.
	if report.DeletionTimestamp == nil && isReportExpired(logger, report, time.Now() {
```

Reply:
I guess you would lose the `op.logger.Warnf(...)` call, but there are probably better ways to structure this anyways.

Reply:
There are better ways to structure, but next pass...